### PR TITLE
add gzip compression and code-splitting example

### PR DESCRIPTION
**Pull request by krazyjakee**

### CODE REVIEW

**Comment by krazyjakee** [comment]
Ok, this is now a complete and working code-split, SSR solution supported by https://github.com/theKashey/react-imported-component

I had to add the parcel favicon to clean up network requests in the browser.

**Comment by eXon** [comment]
Just wow... @krazyjakee  you did an amazing job! I never tried code splitting with SSR because of the complexity of mixing both together but it looks like you nailed it. I'll try your code and give some feedback.

File: app/client.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Start your React application and add the required containers
 // Here we have <BrowserRouter /> for react-router
 
+import {rehydrateMarks, assignImportedComponents} from 'react-imported-component';
```
Comment (eXon) [inline app/client.js]: `assignImportedComponents` is unused

File: app/client.js [hunk 2]
```diff
@@ -15,13 +16,16 @@ const app = (
...
 
-// In production, we want to hydrate instead of render
-// because of the server-rendering
-if (process.env.NODE_ENV === 'production') {
-  ReactDOM.hydrate(app, element);
-} else {
-  ReactDOM.render(app, element);
-}
+// rehydrate the bundle marks
+rehydrateMarks().then(() => {
```
Comment (eXon) [inline app/client.js]: `rehydrateMarks()` could have been used for production only. Would make the code more readable.

**Comment by krazyjakee** [comment]
- [x] generate imports.js
- [x] add generate-imported-component command
- [x] remove unused assignImportedComponents import
- [x] only use rehydrateMarks in production

**Comment by krazyjakee** [comment]
Also added the airbnb eslint configuration with a couple of tweaks to make it work with the project.

**Comment by eXon** [comment]
@krazyjakee This is great! However, I don't want something as opiniated as the `airbnb` eslint config in here. Just the eslint/react recommended would be enough.

The readme should also be updated with more details about the changes.

**Comment by krazyjakee** [comment]
@eXon thanks! I reset eslint config to defaults and updated the readme a little bit. Not sure what more to add, any suggestions?

**Comment by eXon** [comment]
@krazyjakee Add your `HelloWorld2.jsx` in the `Read the code` section and it will be all good :)

For eslint, you can set `env.browser` and `env.node` instead of adding manually the global variables. You should also add this because no rule is enabled by default:
```json
{
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ]
}
```

Thanks!

**Comment by krazyjakee** [comment]
Added helloworld2 and corrected the eslintrc file

**Comment by RaulTsc** [comment]
@krazyjakee need help with anything? :)

**Comment by eXon** [comment]
Thanks @krazyjakee for the contribution :D

**Pull request merged**